i_garantie_button_implementieren

joeymez opened the pull request:
Fertigstellung des Items I_Garantie_Implementieren: 
----------------------------------------------------------

-> Hinzufügen eines Icons zur Behandlung der Garantie
-> Hinzufügen eines neuen Icons zum Speichern
-> Änderung des Buttons *Speichern* zu einem ImageButton (Icon)
-> Anpassung der beiden neuen Icons an die Toolbar + Material Design
-> Erstellung zweier PopUps zur Behandlung des GarantieButtons
     (jeweils wenn Garantie aktiviert wird und wenn Garantie deaktiviert wird)

-> Hinzufügen von neuen Strings in der Strings-XML für die PopUpFenster

Commit:
Fertigstellung des Items I_Garantie_Implementieren:

-> Hinzufügen eines Icons zur Behandlung der Garantie
-> Hinzufügen eines neuen Icons zum Speichern
-> Änderung des Buttons *Speichern* zu einem ImageButton (Icon)
-> Anpassung der beiden neuen Icons an die Toolbar + Material Design
-> Erstellung zweier PopUps zur Behandlung des GarantieButtons
     (wenn Garantie aktiviert und wenn Garantie deaktiviert wird)
-> Hinzufügen von neuen Strings in der Strings-XML für die PopUpFenster

Commit:
Update des Items I_Garantie_Implementieren:
--------------------------------------------------------------

-> Verschieben der Icons im *Drawable* Ordner in den *MipMap* Ordner
-> Änderung des Pfades in der XML für den Zugriff auf die verschobenen Icons

Commit:
Update & Verbesserung des Items I_Garantie_Implementieren:
--------------------------------------------------------------

ImageButtons:
-> Image Button (Garantie) verändert die Farbe sobald es aktiviert wird, d.h. wenn Garantie-Button gedrückt wird
-> bei erneutem drücken des Garantie-Buttons wird die ursprüngliche Farbe angenommen (Weiß)

PopUp Fenster:
-> neues PopUp-Fenster (INFO) wurde erstellt ... dieses schließt sich automatisch nach der im Parameter angegebenen Zeit
-> wird jeweils beim drücken des ImageButtons *Garantie* aktiviert
-> Einbindung dieser neuen Methode als STATIC Methode in die Statische Klasse

String-XML:
-> löschen der Strings, die nicht mehr gebraucht werden

Commit:
Update & Verbesserung des Items I_Garantie_Implementieren:
--------------------------------------------------------------

ImageButtons:
-> Image Button (Garantie) verändert die Farbe sobald es aktiviert wird, d.h. wenn Garantie-Button gedrückt wird
-> bei erneutem drücken des Garantie-Buttons wird die ursprüngliche Farbe angenommen (Weiß)

PopUp Fenster:
-> neues PopUp-Fenster (INFO) wurde erstellt ... dieses schließt sich automatisch nach der im Parameter angegebenen Zeit
-> wird jeweils beim drücken des ImageButtons *Garantie* aktiviert
-> Einbindung dieser neuen Methode als STATIC Methode in die Statische Klasse

String-XML:
-> löschen der Strings, die nicht mehr gebraucht werden

Sippl commented on app/src/main/java/de/projektss17/bonpix/S.java:
```diff
@@ -144,6 +150,41 @@ public void onClick(DialogInterface dialog, int which) {
+
+    public static void popUpInfo(View v, int title, int message, int time){
+
+        AlertDialog.Builder builder = new AlertDialog.Builder(v.getContext());
```
wieso übergibst du hier eine View statt wie in der popUpDialog "AppCompatActivity" und holst dir aus dieser den Context der App? Dazu machst du gerade genau das gleiche wie in der popUpDialog nur das du hier keine Negative- und Positive Buttons hinzufügst. Bitte bau das noch einheitlich damit wir hier eine Linie haben. Zusätzlich ist mir aufgefallen das die Farbe des Alerts anders ist als in der popUpDialog, kannst du diese bitte auch noch weiß machen?  danke :slightly_smiling_face:

joeymez replied:
Nunja das ist ganz einfach erklärt! Ich habe in diesem Fall eine View benutzt, da sich die Farbe des Fenster dementsprechend in dunkel grau einfärbt.... Da es sich hier um ein InfoFenster handelt, dass sich selbst schließt, wollte ich die Farbe anders haben.

-> Wenn euch die Farbe stört ändere ich sie ab  ....
-> werde es erneut commiten!
-> bitte erneut checken und gegebenfalls ok geben

Sippl reviewed (changes requested):
Hallo Johanns, deine Umsetzung mit dem Button und das dieser ausgegraut wird, finde ich echt gelungen. Die Anmerkungen für meinen Change Request findest du in der Code Review.

Commit:
Update & Ausbessern des Items I_Garantie_Implementieren:
--------------------------------------------------------------

PopUp Fenster:
- Änderung eines Übergabeparameters
- DialogFenster nun an den Speicher DialogFenster angepasst (> Farbe weiß)

dontcallme commented on app/src/main/res/layout/box_ocr_manuell_screen.xml:
```diff
@@ -28,22 +31,54 @@
+                    android:layout_marginEnd="16dp"
+                    android:layout_marginRight="16dp"
+                    android:background="@color/colorPrimary"
+                    app:srcCompat="@mipmap/ic_save_button"
```
Wegen der einheitlichkeit, hier bitte ic_save_button_24dp verwenden / mipmap anpassen

joeymez replied:
wenns sein muss xD

Sippl replied:
Hi Johanns, hast du dies was Dominik gemeint hat hier angepasst? Weil bei mir auf dem Handy erscheint das Icon ic_verified_user_24dp ein klein bisschen größer als das SAVE? Falls es so gehört und du ic_save_button bereits auf 24 dp geändert hast, dann vergiss bitte meine Aussage :D. Außerdem wäre es gut, wenn du bitte den Namen der Mipmap auf ic_save_button_24dp umzubenennen. Dann ist es einheitlich :)

dontcallme commented on app/src/main/java/de/projektss17/bonpix/A_OCR_Manuell.java:
```diff
@@ -42,14 +42,16 @@
     private View mExclusiveEmptyView;
     private EditText anschriftInput;
     private LinearLayout linearLayout;
+    private ImageButton garantieButton, saveButton;
```
Ist hier wirklich ein neuer Typ ImageButton notwendig oder kann nicht einfach Button mitbenutzt werden?

joeymez replied:
Hi domi! 
Ja ein ImageButton ist in diesem Fall notwendig. Ich habe es nicht implementiert um mehr Code zu produzieren, sondern vielmehr da ein BUTTON und ein IMAGE-BUTTON komplett verschiedene Dinge sind ^^

- ein Image-Button ist ein Icon auf welches man drücken kann
- ein Button ist ein Textfeld auf welches man drücken kann

dontcallme replied:
Jo da hast du recht ;), hatte das eher auf die Funktion bezogen, die ja dieselbe ist. Aber kein Problem, ist ja nur 1 Zeile^^

dontcallme reviewed (changes requested):
Hi Johanns, sieht soweit alles recht gut aus, ein paar Anmerkungen hab ich dir im Code noch dagelassen.

-Über die Farbe des Garantiesymbols bin ich mir noch  nicht schlüssig, kann man aber erstmal so lassen.
- Popupfenster ist soweit OK, allerdings hast du einmal INFO und einmal Hinweis verwendet --> würde es einheitlich machen

Sonst gute Arbeit, funktioniert alles :)

Commit:
Update / Verbesserung des Items i_garantie_button_implementieren:
---------------------------------------------------------------------------

- Ändern des Info-Fensters (verwendung eines Toast anstatt eines AlertDialog)
- Entfernen der unnötigen Strings

Sippl reviewed (commented):
Soweit funktioniert bei mir alles. Sieht super aus Johanns!!! Einzig der Change Request vom Domi mit dem Icon muss noch angepasst werden (Falls du es noch nicht hast). Danach kann ich grünes Licht geben. Wir sollten uns im Hinterkopf behalten, dass in den nächsten Sprints die Funktion jedoch noch erweitert werden muss (Damit Garantie-Ende Datum via PopUp bei Garantie hinzufügen, mitgepflegt werden kann). Keine Sorge, das jetzige Item beinhaltet noch nicht diese Anforderung (wurde bei Item Erstellung vergessen...).

Commit:
Update i_garantie_button_implementieren:
-----------------------------------------------

- Umbennen des Icons i_save_button_24dp und hinzufügen dieses Icons ins Git

Sippl reviewed (approved):
Alle Änderungswünsche angepasst, funktioniert. Von meiner Seite passt es.

dontcallme reviewed (approved):
Jetzt passt alles, von mir gibts ein OK.
Sehr Gut :)
